Orange/data/table/base.py

```diff
@@ -0,0 +1,1383 @@
+                # domain must contain the uniques of all variables
+                new_domain = Domain(
+                    unique_by_name_preserve_order(flatten(t.domain.attributes for t in tables)),
+                    unique_by_name_preserve_order(flatten(t.domain.class_vars for t in tables)),
+                    unique_by_name_preserve_order(flatten(t.domain.metas for t in tables))
+                )
+            new_index = cls._new_id(len(result))
+            result.index = new_index
+        elif axis == CONCAT_COLS:
+            # check for same name
+            columns = [v for v in flatten([list(t.columns) for t in tables]) if v not in cls._INTERNAL_COLUMN_NAMES]
+            if len(set(columns)) != len(columns):
+                raise ValueError("Cannot concatenate domains with same names.")
+            if colstack:
+                # check for same length
+                if len(set(len(t) for t in tables)) != 1:
+                    raise ValueError("Cannot colstack tables with differing numbers of rows. ")
+                # reset index temporarily because this joins by index by default
+                result = pd.concat([t.reset_index(drop=True) for t in tables], axis=1, join_axes=[tables[0].index])
+            else:
+                result = pd.concat(tables, axis=1)
+
+            # merge domains
+            new_domain = Domain(
+                list(flatten(t.domain.attributes for t in tables)),
+                list(flatten(t.domain.class_vars for t in tables)),
+                list(flatten(t.domain.metas for t in tables))
+            )
+
+            # fix multiple weight columns: keep the first one
+            weight_columns = result[cls._WEIGHTS_COLUMN]
+            result = result.drop(cls._WEIGHTS_COLUMN, axis=1)
+            result[cls._WEIGHTS_COLUMN] = weight_columns.iloc[:, 0]
+
+            if reindex:
+                new_index = cls._new_id(len(result))
+                result.index = new_index
+        else:
+            raise ValueError('axis {} out of bounds [0, 2)'.format(axis))
+        tmpdomain = result.domain
+        result.__finalize__(tables[0])  # pd.concat does not do this by itself
+        result.domain = tmpdomain  # we don't want to transfer the domain, specifically
+        return cls.from_dataframe(new_domain, result)
+
+    def merge(self, right, *args, **kwargs):
+        """Merge two Tables. pd.DataFrame.merge wrapper.
+
+        Handles internal columns and domain merging. Renames duplicates appropriately.
+
+        Parameters
+        ----------
+        right : TableBase
+            The other Table to merge.
+        args
+        kwargs
+
+        Returns
+        -------
+        TableBase
+            A new, merged Table.
+
+        See Also
+        --------
+        pd.DataFrame.merge
+        """
+        # rename default suffixes
+        suffix_left, suffix_right = kwargs.get('suffixes', ('_left', '_right'))
```

Comment:
Okay. I added it to the signature so I don't have to manually override in the method itself, it's cleaner that way I think.